pkg/k8sutil/resource.go

```diff
@@ -133,8 +134,20 @@ func Reconcile(log logr.Logger, client runtimeClient.Client, desired runtime.Obj
 			case *corev1.ConfigMap:
 				// Only update status when configmap belongs to broker
 				if id, ok := desired.(*corev1.ConfigMap).Labels["brokerId"]; ok {
-					currentConfigs := util.ParsePropertiesFormat(current.(*corev1.ConfigMap).Data[kafka.ConfigPropertyName])
-					desiredConfigs := util.ParsePropertiesFormat(desired.(*corev1.ConfigMap).Data[kafka.ConfigPropertyName])
+					currentConfigs, err := properties.NewFromString(current.(*corev1.ConfigMap).Data[kafka.ConfigPropertyName])
+					if err != nil {
+						return errors.WrapIf(err, fmt.Sprintf("could not parse the current configuration for broker with id: %s", id))
```

Comment:
Instead of `errors.WrapIf(err, fmt.Sprintf())`  `errors.WrapIff()` can be used. Thinking it more the use structured error would be even better `errors.WrapWithDetails(err, "could not parse the current configuration for broker", "borkerId", id)`

Reply:
Agreed. Will fix. Ty.